src/shogun/base/SGObject.cpp

```diff
@@ -733,15 +733,25 @@ void CSGObject::create_parameter(
 
 void CSGObject::update_parameter(const BaseTag& _tag, const Any& value)
 {
-	if (!self->map[_tag].get_properties().has_property(
+	auto any_param = self->map[_tag];
+	auto pprop = self->map[_tag].get_properties();
+	if (!pprop.has_property(
 	        ParameterProperties::READONLY))
 		self->update(_tag, value);
 	else
 	{
 		SG_ERROR(
-		    "%s::%s is marked as read-only and cannot be modified", get_name(),
+		    "%s::%s is marked as read-only and cannot be modified!\n", get_name(),
 		    _tag.name().c_str());
 	}
+	if (pprop.has_property(ParameterProperties::CONSTRAIN))
+	{
+		if (any_param.get_constrain_function()(value))
+			self->update(_tag, value);
+		else
+			SG_ERROR("%s::%s cannot be updated because of its constraint!\n", get_name(),
```

Comment:
the error message could be obtained from the constrain_functions which would have some virtual function with a message, e.g. positive would have "requires value to be positive"